src/sage/matroids/chow_ring_ideal.py

```diff
@@ -344,6 +374,480 @@ def normal_basis(self, algorithm='', *args, **kwargs):
         return PolynomialSequence(R, [monomial_basis])
 
 
+# Redirecting the unpickling of old class:
+from sage.misc.persist import register_unpickle_override
+register_unpickle_override('sage.rings.chow_ring_ideal', 'ChowRingIdeal_nonaug_fy', ChowRingIdeal_nonaug_fy)
```

Comment:
```suggestion
register_unpickle_override('sage.rings.chow_ring_ideal', 'ChowRingIdeal_nonaug', ChowRingIdeal_nonaug_fy)
```
This needs to redirect the old name into the new name. You can check this by getting the old pickle `dumps(chow_ring_ideal)` on `develop` and then `loads(pickle_string)` with the new code/branch.

Reply:
I made this correction, as well as the one from `sage.rings.chow_ring_ideal` to `sage.matroids.chow_ring_ideal`. However the loading of the old pickle fails. Does it work for you?

Reply:
In vanilla Sage:
```
sage: M = matroids.Uniform(2,4)
sage: I = M.chow_ring(QQ).defining_ideal()
sage: dumps(I)
b'x\x9c\xadVgx\xdc\xc6\x11=\x92\xa2\n$\xb9\xc5Jb\xcbI\x10\xa5\x1d#\xfa"RJSh\'\x8e\\\xa2\x9cH;\x139A\x12Q8\x1cny@\x88\x03\xee\xa1H\xa4\x99\x8b\xed$$\xe3\xf4\xde{\xef\xbd\xf7\xde{\xef\xbd\xf7\xde\xdb\xec\x0ex\xe2\xf9\x93\xfe\xe5\xbb\xef\xdb\xdb\xc5\xbc\xf7fvvv\x80k\x86\xfd\xcck\xabZ\xc7\xcb\xd3$le5?H\x8e\xbbi\x18\xb7\xdd\xb0\xa5\xbc\xc8:\xc0k\xe2\xe5A\xbdr\xe3$\xf6\x8a\xb6\x85\xca\xd8\xb5\x18\x12f\x96\xa7\x85\x9f\x17\xa9\xaa\xb5\x16c\xaf\x13\xfa\xae\x1fyYf\r\xae0\\uvV*\x15#3\x9d\xc4\xec\xcb\xf5\xdd\xe3a\x1e\xb8\xbe\x97\xabv\x92.bD\x04\xb5\xf3\xacf\xbc\xbb\x1d\x83\xb4\x06X\x166\xf8\xae\xdb,\xc2(\x0fc\xd7\xb5\xda*\xf7\xf2<\xb50Zn%\xcc\xfc\x9a\xefE\x91\xa5\x07\xb7\xa3\xf2 iY\xd8\x18\xf0\xef\xfa!\x17q\x88B\xb9\xa9\xea\xa6*S1\x0b\x85Il\x15q\xaaZ\x85\xaf,l\x12B\x19b\xa82NT\xbb\xe3e\xd6\xb4\xfca\xf3X\x0f[Va\x11\xb6:[y\x7f\xb2%o!L:\xd8f\x9e\\\x94e\x89\x1f\xb2\xf01\x85\xed\xab8\x8dpz\xb0\xcd\xd9\xc8\x96+\xe30\xf7"\x9c\xb1\x8a3\tg\x05\x02?\x90t:E.\xf0\x1b\xac\xe2l\xc2\x0eg\x1b?\xefz)\xc7\'\xd9\xc4\rWp#\xc2\x8dWp\xce\xccL\xb0!\xc7\xb9\x84\x9d|"\xe7\xf5p\x93\xaas\xa6\x0ec0\xcf.\xe1\xa6\xeb\x93\xdbM\xa2\xc58\xe9\x84^T\xeb\x14\x9cF\xf7\xc4\x037\n\x9b\x19\x83\x8a\xc8K9\x11\xdd\xd0\x9f\x8f\x94;}E\x1f\xa0\x8ba\x00\x84\x9b\xad\x97NM\nYg.TQ\xcb\xa2ry\xa9Y\xc1\x1e#\xdc\xbc\xea\x0c\xeb\xbc\xec\xc1.\x99L\xe0\x162\x99\xc4-e\xb2\x17\xb7rF\rfbr/n\x9d\xe36\xa7\x88>Wi\xc7M\xd2\x96J\xad\xc3<\xbd\xdc\xccP\xe5\\\xf0\xb9\xdc\xb6\xeal\xd2\xb9hF\x89?\x9fa\xf7\x98\xb3\xd9\x9c\x90\n\xdbA\x9ea|F\xcc\xba\xf4\xc3\x05\x9c?\xe3\x9c\xad\x97k\xfb\xe2\xdaKUI\xad\xd5+\xce\xb9\x03F\x1d\x89\xf1\xeb\xfaITtb\xdc\xaeT\x8bT\xdc\xce\x03\xec\xa9\x8f\x9a\x1d\xb8|\t\x14&\x9c-<o\xa9v\xaa\x8eEj\x01\x93\xce\xf6\x015.H\xec5[ou\xb1\xcf9M\xc2\xf2=6.N\x1a\xeb\xed\x8d\xfae\xa4\x8e\x1db\xfe\x1d\x1cK \\\x84\xc6|G\xe3`W\xa9\xbf\x0bw\x92`\xdcV\xc2\x87\x8f;;g\rU*\xd6\xc5\xda\xbf\xb2\x19\xa3\xd2L\xd9\x0c\x0c\xfd\xa4\x9dz\xdd \xf4\xedj?\xba1[g\xd56\xbb\xabY\xd6!\x95\xdb\x8d#l\xad.\x1c\xf5\xc6\xec\x0bl\xcf\x9d\xb0w\xf38\xc9\xe3\x91V\x92gf\x157\xc6\xed<P\xb1\xd5`\x98=e/\x1cm6\xecp\xce\xf6\xe2\x96\x9d\xc4\xd1\xa2\x9e\xaf\x93\x99\xb2\xcbys\xac\xc1\xae\x06=\xac31\xddb\xd9T\xd9\x1cm\xc6\xbe\x1a\x13\xf6\x91H\xd9\xa1\x19\xe3\x86\x9d\x15~\xc0\x8e=\x0e\x92\x83`r\xd3\x8d\xc7%\xaeq\x0ek)\xdc=\xd13O\xcdL?\n\xed\x0by\x196j\x16\xf6\x17\xcdU\xdc\x850e\xd2w\xa2\x13]\x10\xec\x90\xd3kz\x99\xc2\x85\x03})f\x8c\x92\x02\xb0\x0e\xcaB_\t\x0bw\xe5\xda\xbe\x9b\xb9\xd8\xe6\xd03\\4\xe3\x9c\xaeW\xe5\x1d\xd2Y\xe7\xbb\x80\xbb\xd7\x87\xe4\xf8T\xa7\xa9Z-\xe6\xe2@Y{*R\x1ds\xcb\x93XzT\x92\xe2\xe2u]mN\xa4\xfa\xf7r\xba\xecm\x978;NAwqip^\xb0s\x15\x97\x11\xee!N\xd8\xca\xa1\xe4ec\x94\x02=8\xe3\x9c\xb3^!\xe4\xd6\xe4v\xb9\xd5\xb8\xd2wp\xcf\xeb\x84\xac\r\xbaK\xf8\x89J}\xc5} \xcbQ\x9f\xc5\xa1A\xab\xe7\xeb[/\xd6\xe9Y\xccHx\'\xb8\x12\x801_>\x8b+\x8af\x0f\xf7\xaa\xca\xed,_D\xa0\xeb\xbd\x97\xca-\xebL\xf7\xbb\x92\x1f\xa6~\x11\xea\xa6\x98d\xdc\xcd\xb35\xb2\x85{\xd7+\x03\xef\x88\xb94\xb9J\xc5\x99\xca-\x1c\x9e\xc5\x95\xd5z\xa5>T\x1f\xae\x8f\xa8e\xdc\x87p\xdf\x1e\x9c\x93\x9c\xb1\xd5\xf1\xe6\x95\xbb\xb6\xc0\xfd\x1c\xbeD\x95I\xdc\x7f\x19\x0f \x1c\tXhV\x0fG\xd7\xabq\xa7mx\xcb\xf0\x08\xcd\xcc!\xdd\xe4\xab\x93\xe3\xf6\xbe\xb1\xfd\xf6\xb4\xc4f\'sv\xea\xc5\xf3\xf6$_\x0b{\x9f]f<\xb3\xf5\x8b\xc3.wt\xfe\xda\x8e\xac\xa5\xc9\xfd\xf6\xd2\xd2\x9eq{b\xdcf\xa1\xbd\xbd^\x0f\xfe*Z+P\x849\xe7\x0c\x9d\xb2\xb9\xc8\xcb3\xb7\xadb\xc5\xcd\x97+\xa6\xddCP\xd5\x91\x85\xf5\n\xc7\xf2@\xc2\xfc\xff\xab\xf9\x0f6\xfe(\x98\xea\xa1s\nm=m\x85~n]r\xb8\xe8r\xcd"\xee!\xd1\x99\xca\xea\xa3+\xe8\x12p\xb2\xd7\x86\xe4\xbc\xbfBj\x92>\x81l\x199\xa1\xc8Vp\x8cp\\on\xa1>\xc4\x9b[$\\\x15\x988\x96\x02v\xf0 >\x08q\xd0#<8H\x03\xe6]M\xb8\x86y\xd7\x12\x1e\xa2y\x0f\xad\x0f3\xefa\x84e\xe1\xadh\xde*\x1f\x9f\xf0\x1eN\xb8Nx\x8f<\x92y\x8f"<Z\xf3\x1eS\x1fa\xdec\t\x8f\x13\xde\xe35\xef\t\xf5\x91\x92\xf7D\xc2\x93\x84\xf7d\xc2S\x98\xf7T\xc2\xd34\xef\xe9\xeb\xcb\xe3\x19\x84g\n\xfdY\x9a\xfe\xec\xfa\x86\x92\xfe\x1c\xc2s\x85\xfe<\xc2\xf3\x99\xfe\x02\xc2\x0b\x0b\xb9U\xf2\xe9e\xce\x97\xbf\xa7L\xe7\xc1\x8b\x82\xa9\x93\x1ay\x92\xe1\xc5U\xf1\xf1\x12\xed\xe3\xa5\xe2\x9f\x7f\x85v\xf42\xc2\xcb\xc5\xd1+\x08\xafdG\xaf"\xbcZ\xe0\xaf\xd1\xf0\xd7\xae\x85+\xf0\xd7\x11^/\xf07\x10\xde\xc8\xf07\x11\xde,\xf0\xb7h\xf8[\x05>R\xc2\xdfFx\xbb\xc0\xdfAx\'\xc3\xdfEx\xb7\xc0\xdf\xa3\xe1\xef\xad\x9aP\xd6\xd4\xdfGx\xbf\xc0?@\xf8 \xc3?D\xf8\xb0\xc0?\xa2\xe1\x1f\x15\xf8\x9a\xfa\xc7\x08\x1f\x17\xf8\'\x08\x9fd\xf8\xa7\x08\x9f\x16\xf8g4\xfc\xb3U\xad\xdc\x87\x7f\x8e\xf0y\x81\x7f\x81\xf0E\x86\x7f\x89\xf0e\x81\x7f\xa5\xaa\xf1_\xed\x97\xe3\xd7\x08_\x17\xe87\x08\xdf\xd4\xb6o\xf5\x8f\xe6\xdb\x84\xef\x88\xed\xbb\x84\xef\x15+\xf8>\xe1\x07"\xf3C#\xf3\xa3~\xd1\xfd\x98\xf0\x13\x81\xfe\x94\xf03m\xfby_\xe6\x17\x84_\x8a\xedW\x84_\xb3\xcco\x08\xbf\x15\x99\xdf\x19\x99\xdf\xf7k\xf0\x0f\x84?\n\xf4O\x84?k\xdb_\xfa2\x7f%\xfcMl\x7f\'\xfc\x83e\xfeI\xf8\x97\xc8\xfc\xdb\xc8\xfc\xa7_\x92\xff\xa5\xb4\xc2\xb7\xc7\xa0\xd3!\x9eQ:\xac\xd7q/\x1d\xe1\xff5\xc9t\x831\x8d\xf6\xa1\x1b\xcdz\x13\x8f\xc5J\xba\xd9,\xb6\xf0\xc8\x9f\xd8<\xca+\xad\xddt\x9b\x8b\xf2\xe1\xc5\xd5\x98neC/\xdd&\x8c\xed\xfc\xd7\xac\xfd\x0fQ\xfe]\x94'
```
Then with your PR:
```
sage: loads(b'x\x9c\xadVgx\xdc\xc6\x11=\x92\xa2\n$\xb9\xc5Jb\xcbI\x10\xa5\x1d#\xfa"RJSh\'\x8e\\\xa2\x9cH;\x139A\x12Q8\x1cny@\x88\x03\xee\xa1H\xa4\x99\x8b\xed$$\xe3\xf4\xde{\xef\xbd\xf7\xde{\xef\xbd\xf7\xde\xdb\xec\x0ex\xe2\xf9\x93\xfe\xe5\xbb\xef\xdb\xdb\xc5\xbc\xf7fvvv\x80k\x86\xfd\xcck\xabZ\xc7\xcb\xd3$le5?H\x8e\xbbi\x18\xb7\xdd\xb0\xa5\xbc\xc8:\xc0k\xe2\xe5A\xbdr\xe3$\xf6\x8a\xb6\x85\xca\xd8\xb5\x18\x12f\x96\xa7\x85\x9f\x17\xa9\xaa\xb5\x16c\xaf\x13\xfa\xae\x1fyYf\r\xae0\\uvV*\x15#3\x9d\xc4\xec\xcb\xf5\xdd\xe3a\x1e\xb8\xbe\x97\xabv\x92.bD\x04\xb5\xf3\xacf\xbc\xbb\x1d\x83\xb4\x06X\x166\xf8\xae\xdb,\xc2(\x0fc\xd7\xb5\xda*\xf7\xf2<\xb50Zn%\xcc\xfc\x9a\xefE\x91\xa5\x07\xb7\xa3\xf2 iY\xd8\x18\xf0\xef\xfa!\x17q\x88B\xb9\xa9\xea\xa6*S1\x0b\x85Il\x15q\xaaZ\x85\xaf,l\x12B\x19b\xa82NT\xbb\xe3e\xd6\xb4\xfca\xf3X\x0f[Va\x11\xb6:[y\x7f\xb2%o!L:\xd8f\x9e\\\x94e\x89\x1f\xb2\xf01\x85\xed\xab8\x8dpz\xb0\xcd\xd9\xc8\x96+\xe30\xf7"\x9c\xb1\x8a3\tg\x05\x02?\x90t:E.\xf0\x1b\xac\xe2l\xc2\x0eg\x1b?\xefz)\xc7\'\xd9\xc4\rWp#\xc2\x8dWp\xce\xccL\xb0!\xc7\xb9\x84\x9d|"\xe7\xf5p\x93\xaas\xa6\x0ec0\xcf.\xe1\xa6\xeb\x93\xdbM\xa2\xc58\xe9\x84^T\xeb\x14\x9cF\xf7\xc4\x037\n\x9b\x19\x83\x8a\xc8K9\x11\xdd\xd0\x9f\x8f\x94;}E\x1f\xa0\x8ba\x00\x84\x9b\xad\x97NM\nYg.TQ\xcb\xa2ry\xa9Y\xc1\x1e#\xdc\xbc\xea\x0c\xeb\xbc\xec\xc1.\x99L\xe0\x162\x99\xc4-e\xb2\x17\xb7rF\rfbr/n\x9d\xe36\xa7\x88>Wi\xc7M\xd2\x96J\xad\xc3<\xbd\xdc\xccP\xe5\\\xf0\xb9\xdc\xb6\xeal\xd2\xb9hF\x89?\x9fa\xf7\x98\xb3\xd9\x9c\x90\n\xdbA\x9ea|F\xcc\xba\xf4\xc3\x05\x9c?\xe3\x9c\xad\x97k\xfb\xe2\xdaKUI\xad\xd5+\xce\xb9\x03F\x1d\x89\xf1\xeb\xfaITtb\xdc\xaeT\x8bT\xdc\xce\x03\xec\xa9\x8f\x9a\x1d\xb8|\t\x14&\x9c-<o\xa9v\xaa\x8eEj\x01\x93\xce\xf6\x015.H\xec5[ou\xb1\xcf9M\xc2\xf2=6.N\x1a\xeb\xed\x8d\xfae\xa4\x8e\x1db\xfe\x1d\x1cK \\\x84\xc6|G\xe3`W\xa9\xbf\x0bw\x92`\xdcV\xc2\x87\x8f;;g\rU*\xd6\xc5\xda\xbf\xb2\x19\xa3\xd2L\xd9\x0c\x0c\xfd\xa4\x9dz\xdd \xf4\xedj?\xba1[g\xd56\xbb\xabY\xd6!\x95\xdb\x8d#l\xad.\x1c\xf5\xc6\xec\x0bl\xcf\x9d\xb0w\xf38\xc9\xe3\x91V\x92gf\x157\xc6\xed<P\xb1\xd5`\x98=e/\x1cm6\xecp\xce\xf6\xe2\x96\x9d\xc4\xd1\xa2\x9e\xaf\x93\x99\xb2\xcbys\xac\xc1\xae\x06=\xac31\xddb\xd9T\xd9\x1cm\xc6\xbe\x1a\x13\xf6\x91H\xd9\xa1\x19\xe3\x86\x9d\x15~\xc0\x8e=\x0e\x92\x83`r\xd3\x8d\xc7%\xaeq\x0ek)\xdc=\xd13O\xcdL?\n\xed\x0by\x196j\x16\xf6\x17\xcdU\xdc\x850e\xd2w\xa2\x13]\x10\xec\x90\xd3kz\x99\xc2\x85\x03})f\x8c\x92\x02\xb0\x0e\xcaB_\t\x0bw\xe5\xda\xbe\x9b\xb9\xd8\xe6\xd03\\4\xe3\x9c\xaeW\xe5\x1d\xd2Y\xe7\xbb\x80\xbb\xd7\x87\xe4\xf8T\xa7\xa9Z-\xe6\xe2@Y{*R\x1ds\xcb\x93XzT\x92\xe2\xe2u]mN\xa4\xfa\xf7r\xba\xecm\x978;NAwqip^\xb0s\x15\x97\x11\xee!N\xd8\xca\xa1\xe4ec\x94\x02=8\xe3\x9c\xb3^!\xe4\xd6\xe4v\xb9\xd5\xb8\xd2wp\xcf\xeb\x84\xac\r\xbaK\xf8\x89J}\xc5} \xcbQ\x9f\xc5\xa1A\xab\xe7\xeb[/\xd6\xe9Y\xccHx\'\xb8\x12\x801_>\x8b+\x8af\x0f\xf7\xaa\xca\xed,_D\xa0\xeb\xbd\x97\xca-\xebL\xf7\xbb\x92\x1f\xa6~\x11\xea\xa6\x98d\xdc\xcd\xb35\xb2\x85{\xd7+\x03\xef\x88\xb94\xb9J\xc5\x99\xca-\x1c\x9e\xc5\x95\xd5z\xa5>T\x1f\xae\x8f\xa8e\xdc\x87p\xdf\x1e\x9c\x93\x9c\xb1\xd5\xf1\xe6\x95\xbb\xb6\xc0\xfd\x1c\xbeD\x95I\xdc\x7f\x19\x0f \x1c\tXhV\x0fG\xd7\xabq\xa7mx\xcb\xf0\x08\xcd\xcc!\xdd\xe4\xab\x93\xe3\xf6\xbe\xb1\xfd\xf6\xb4\xc4f\'sv\xea\xc5\xf3\xf6$_\x0b{\x9f]f<\xb3\xf5\x8b\xc3.wt\xfe\xda\x8e\xac\xa5\xc9\xfd\xf6\xd2\xd2\x9eq{b\xdcf\xa1\xbd\xbd^\x0f\xfe*Z+P\x849\xe7\x0c\x9d\xb2\xb9\xc8\xcb3\xb7\xadb\xc5\xcd\x97+\xa6\xddCP\xd5\x91\x85\xf5\n\xc7\xf2@\xc2\xfc\xff\xab\xf9\x0f6\xfe(\x98\xea\xa1s\nm=m\x85~n]r\xb8\xe8r\xcd"\xee!\xd1\x99\xca\xea\xa3+\xe8\x12p\xb2\xd7\x86\xe4\xbc\xbfBj\x92>\x81l\x199\xa1\xc8Vp\x8cp\\on\xa1>\xc4\x9b[$\\\x15\x988\x96\x02v\xf0 >\x08q\xd0#<8H\x03\xe6]M\xb8\x86y\xd7\x12\x1e\xa2y\x0f\xad\x0f3\xefa\x84e\xe1\xadh\xde*\x1f\x9f\xf0\x1eN\xb8Nx\x8f<\x92y\x8f"<Z\xf3\x1eS\x1fa\xdec\t\x8f\x13\xde\xe35\xef\t\xf5\x91\x92\xf7D\xc2\x93\x84\xf7d\xc2S\x98\xf7T\xc2\xd34\xef\xe9\xeb\xcb\xe3\x19\x84g\n\xfdY\x9a\xfe\xec\xfa\x86\x92\xfe\x1c\xc2s\x85\xfe<\xc2\xf3\x99\xfe\x02\xc2\x0b\x0b\xb9U\xf2\xe9e\xce\x97\xbf\xa7L\xe7\xc1\x8b\x82\xa9\x93\x1ay\x92\xe1\xc5U\xf1\xf1\x12\xed\xe3\xa5\xe2\x9f\x7f\x85v\xf42\xc2\xcb\xc5\xd1+\x08\xafdG\xaf"\xbcZ\xe0\xaf\xd1\xf0\xd7\xae\x85+\xf0\xd7\x11^/\xf07\x10\xde\xc8\xf07\x11\xde,\xf0\xb7h\xf8[\x05>R\xc2\xdfFx\xbb\xc0\xdfAx\'\xc3\xdfEx\xb7\xc0\xdf\xa3\xe1\xef\xad\x9aP\xd6\xd4\xdfGx\xbf\xc0?@\xf8 \xc3?D\xf8\xb0\xc0?\xa2\xe1\x1f\x15\xf8\x9a\xfa\xc7\x08\x1f\x17\xf8\'\x08\x9fd\xf8\xa7\x08\x9f\x16\xf8g4\xfc\xb3U\xad\xdc\x87\x7f\x8e\xf0y\x81\x7f\x81\xf0E\x86\x7f\x89\xf0e\x81\x7f\xa5\xaa\xf1_\xed\x97\xe3\xd7\x08_\x17\xe87\x08\xdf\xd4\xb6o\xf5\x8f\xe6\xdb\x84\xef\x88\xed\xbb\x84\xef\x15+\xf8>\xe1\x07"\xf3C#\xf3\xa3~\xd1\xfd\x98\xf0\x13\x81\xfe\x94\xf03m\xfby_\xe6\x17\x84_\x8a\xedW\x84_\xb3\xcco\x08\xbf\x15\x99\xdf\x19\x99\xdf\xf7k\xf0\x0f\x84?\n\xf4O\x84?k\xdb_\xfa2\x7f%\xfcMl\x7f\'\xfc\x83e\xfeI\xf8\x97\xc8\xfc\xdb\xc8\xfc\xa7_\x92\xff\xa5\xb4\xc2\xb7\xc7\xa0\xd3!\x9eQ:\xac\xd7q/\x1d\xe1\xff5\xc9t\x831\x8d\xf6\xa1\x1b\xcdz\x13\x8f\xc5J\xba\xd9,\xb6\xf0\xc8\x9f\xd8<\xca+\xad\xddt\x9b\x8b\xf2\xe1\xc5\xd5\x98neC/\xdd&\x8c\xed\xfc\xd7\xac\xfd\x0fQ\xfe]\x94')
Chow ring ideal of U(2, 4): Matroid of rank 2 on 4 elements with circuit-closures
{2: {{0, 1, 2, 3}}} - non augmented in Feitchner-Yuzvinksy presentation
```
So I don't get any issues. Make sure you put the bit string all on one line with no spaces.

Reply:
Great. It works for me as well. I wasn't loading it properly earlier.